pluginconfig/commit.go

```diff
@@ -130,9 +95,49 @@ type CommitOffchainConfig struct {
 	// This will typically be an arbitrum testnet/mainnet chain depending on
 	// the deployment.
 	PriceFeedChainSelector cciptypes.ChainSelector `json:"tokenPriceChainSelector"`
+
+	// NewMsgScanBatchSize is the number of max new messages to scan, typically set to 256.
+	NewMsgScanBatchSize int `json:"newMsgScanBatchSize"`
+
+	// The maximum number of times to check if the previous report has been transmitted
+	MaxReportTransmissionCheckAttempts uint
+
+	// RMNSignaturesTimeout is the timeout for RMN signature verification.
+	// Typically set to `MaxQueryDuration - e`, where e some small duration.
+	RMNSignaturesTimeout time.Duration `json:"rmnSignaturesTimeout"`
+
+	// RMNEnabled is a flag to enable/disable RMN signature verification.
+	RMNEnabled bool `json:"rmnEnabled"`
+
+	// MaxMerkleTreeSize is the maximum size of a merkle tree to create prior to calculating the merkle root.
+	// If for example in the next round we have 1000 pending messages and a max tree size of 256, only 256 seq nums
+	// will be in the report. If a value is not set we fallback to EvmDefaultMaxMerkleTreeSize.
+	MaxMerkleTreeSize uint64 `json:"maxTreeSize"`
 }
 
-func (c CommitOffchainConfig) Validate() error {
+func (c *CommitOffchainConfig) ApplyDefaults() {
```

Comment:
Nit: does this need to be a public method?

Reply:
Making it private as not used elsewhere